permission_handler_windows/windows/permission_handler_windows_plugin.cpp

```diff
@@ -0,0 +1,193 @@
+    }
+
+    result->Success(EncodableValue((int)PermissionConstants::ServiceStatus::NOT_APPLICABLE));
+    
+  } else if (methodName.compare("checkPermissionStatus") == 0) {
+    result->Success(EncodableValue((int)PermissionConstants::PermissionStatus::GRANTED));
+  } else if (methodName.compare("requestPermissions") == 0) {
+    auto permissionsEncoded = std::get<EncodableList>(*method_call.arguments());
+    std::vector<int> permissions;
+    permissions.reserve( permissionsEncoded.size() );
+    std::transform( permissionsEncoded.begin(), permissionsEncoded.end(),
+                    std::back_inserter( permissions ),
+                    [](const EncodableValue& encoded) {
+                      return std::get<int>(encoded);
+                    });
+    
+    EncodableMap requestResults;
+
+    for (int i=0;i<permissions.size();i++) {
+      auto permissionStatus = PermissionConstants::PermissionStatus::GRANTED;
+      requestResults.insert({EncodableValue(permissions[i]), EncodableValue((int)permissionStatus)});
+    }
+
+    result->Success(requestResults);
+  } else if (methodName.compare("shouldShowRequestPermissionRationale") == 0
+          || methodName.compare("openAppSettings")) {
+    result->Success(EncodableValue(false));
+  } else {
+    result->NotImplemented();
+  }
+}
+
+void PermissionHandlerWindowsPlugin::IsLocationServiceEnabled(std::unique_ptr<MethodResult<>> result) {
+  result->Success(EncodableValue((int)(geolocator.LocationStatus() != PositionStatus::NotAvailable
+        ? PermissionConstants::ServiceStatus::ENABLED
+        : PermissionConstants::ServiceStatus::DISABLED)));
+}
+
+winrt::fire_and_forget PermissionHandlerWindowsPlugin::IsBluetoothServiceEnabled(std::unique_ptr<MethodResult<>> result) {
+  auto btAdapter = co_await BluetoothAdapter::GetDefaultAsync();
+
+  if (btAdapter == nullptr) {
+    result->Success(EncodableValue((int)PermissionConstants::ServiceStatus::DISABLED));
+    co_return;
+  }
+  
+  if (!btAdapter.IsCentralRoleSupported()) {
+    result->Success(EncodableValue((int)PermissionConstants::ServiceStatus::DISABLED));
+    co_return;
+  }
+  
+  auto radios = co_await Radio::GetRadiosAsync();
+
+  for (uint32_t i=0; i<radios.Size(); i++) {
+    auto radio = radios.GetAt(i);
+    if(radio.Kind() == RadioKind::Bluetooth) {
+      co_await radio.SetStateAsync(RadioState::On);
+      result->Success(EncodableValue((int)(radio.State() == RadioState::On
+            ? PermissionConstants::ServiceStatus::ENABLED
+            : PermissionConstants::ServiceStatus::DISABLED)));
+      co_return;
+    }
+  }
+
+  result->Success(EncodableValue((int)PermissionConstants::ServiceStatus::DISABLED));
+}
+
+}  // namespace
+
+void PermissionHandlerWindowsPluginRegisterWithRegistrar(
+    FlutterDesktopPluginRegistrarRef registrar) {
+  PermissionHandlerWindowsPlugin::RegisterWithRegistrar(
+      PluginRegistrarManager::GetInstance()
+          ->GetRegistrar<PluginRegistrarWindows>(registrar));
+}
```

Comment:
nit: please include an empty line at the end of each file.
```suggestion
}

```